### PR TITLE
v17

### PR DESCRIPTION
**Pull request by Zamiell**
Intro and discussion: https://github.com/raineorshine/npm-check-updates/discussions/1346

### CODE REVIEW

**Comment by raineorshine** [comment]
I rebased v17 on main and pushed to [v17-next](https://github.com/raineorshine/npm-check-updates/tree/v17-next) in case you are having trouble with that.

**Comment by raineorshine** [comment]
I've merged `v17-next` into this branch so everything is in one place.

File: README.md [hunk 1]
```diff
@@ -332,7 +332,7 @@ Options that take no arguments can be negated by prefixing them with `--no-`, e.
   </tr>
   <tr>
     <td>--root</td>
-    <td>Runs updates on the root project in addition to specified workspaces. Only allowed with <code>--workspace</code> or <code>--workspaces</code>. (default: false)</td>
+    <td>Runs updates on the root project in addition to specified workspaces. Only allowed with <code>--workspace</code> or <code>--workspaces</code>. (default: true) (default: true)</td>
```
Comment (Zamiell) [inline README.md:335]: bug?

**Comment by raineorshine** [comment]
There is more work to do to upgrade dependencies and convert to esm, but I'm going to get this merged so that we can work off `main`.

Legacy v16 code will be kept on branch [v16](https://github.com/raineorshine/npm-check-updates/tree/v16) for emergency patches.

**Pull request merged**